openedx/features/course_experience/views/course_message.py

```diff
@@ -0,0 +1,130 @@
+        course_start_data = {
+            'course_start_date': format_date(course.start, locale=to_locale(get_language())),
+            'already_started': already_started,
+            'days_until_start_string': days_until_start_string
+        }
+
+        # Register the course home messages to be loaded on the page
+        self.register_course_home_messages(request, course, user_access, course_start_data)
+
+        # Grab the relevant messages
+        course_home_messages = list(CourseHomeMessages.user_messages(request))
+
+        # Return None if user is enrolled and course has begun
+        if user_access['is_enrolled'] and already_started:
+            return None
+
+        # Grab the logo
+        image_src = "course_experience/images/xsy_circle.png"
+
+        context = {
+            'course_home_messages': course_home_messages,
+            'image_src': image_src,
+        }
+
+        html = render_to_string('course_experience/course-message-fragment.html', context)
+        return Fragment(html)
+
+    @staticmethod
+    def register_course_home_messages(request, course, user_access, course_start_data):
+        """
+        Register messages to be shown in the course home content page.
+        """
+        if user_access['is_anonymous']:
+            CourseHomeMessages.register_info_message(
+                request,
+                Text(_(
+                    " {sign_in_link} or {register_link} and then enroll in this course."
+                )).format(
+                    sign_in_link=HTML("<a href='/login?next={current_url}'>{sign_in_label}</a>").format(
+                        sign_in_label=_("Sign in"),
+                        current_url=urlquote_plus(request.path),
+                    ),
+                    register_link=HTML("<a href='/register?next={current_url}'>{register_label}</a>").format(
+                        register_label=_("register"),
+                        current_url=urlquote_plus(request.path),
+                    )
+                ),
+                title=Text(_('You must be enrolled in the course to see course content.'))
+            )
+        if not user_access['is_anonymous'] and not user_access['is_enrolled']:
+            CourseHomeMessages.register_info_message(
+                request,
+                Text(_(
+                    "{open_enroll_link} Enroll now{close_enroll_link} to access the full course."
+                )).format(
+                    open_enroll_link=HTML("<a class='enroll_link' href='#'>"),
+                    close_enroll_link=HTML("</a>")
+                ),
+                title=Text(_(
+                    'Welcome to {course_display_name}'
+                )).format(
+                    course_display_name=course.display_name
+                )
+            )
+        if (user_access['is_staff'] or user_access['is_enrolled']) and not course_start_data['already_started']:
+            CourseHomeMessages.register_info_message(
```

Comment:
Nit: I think we agreed to remove the calendar reminder, or at least make it a text link, until the story where we add the ability to download calendar entries.